systemd/systemd.go

```diff
@@ -197,59 +238,108 @@ func (c *Collector) collect(ch chan<- prometheus.Metric) error {
 	c.logger.Debugf("systemd filterUnits took %f", time.Since(begin).Seconds())
 
 	for _, unit := range units {
-		logger := c.logger.With("unit", unit.Name)
+		c.logger = c.logger.With("unit", unit.Name)
```

Comment:
let's not overwrite logger here, the code was fine here. 

i.e. the logger is supposed to be used only for this unit and shouldn't be affecting struct's field.

Reply:
I'm confused about this one - if we do not overwrite the field, then only the log calls from within `collect(Metric)` will contain the unit field. Logs printed from any of the other functions in `systemd.go` will not contain the unit name, which makes debugging a bit more difficult. Doesn't it make more sense to overwrite the field so all logs send along the unit name they originated from?

Reply:
The basic issue with overriding object's field is that non unit related calls will have last unit's metadata, which is false.

If we are missing unit metadata somewhere then we need to just add 

```
 logger := logger.With("unit", unit.Name)
```

Reply:
👍